### PR TITLE
Migrate docs to next.js

### PR DESCRIPTION
**Pull request by fxOne**
## Checklist

- [x] Fix any eslint errors that occur
- [x] Update change-log for every plugin you touch
- [x] Add/Update tests if you add/change new functionality
- [x] Add/Update docs if you add/change functionality
- [x] Enable "Allow edits from maintainers" for this PR

## Use-case/Problem

Fixes issue #1575

## Implementation

This PR migrates the docs to next.js and solves issues with the old webpack version. I also migrate the docs for `@draft-js-plugins/mention` to TypeScript and hooks. The editor styling should look a little bit different as I could not find the correct Prism theme.

The deploy can be started by calling `yarn deploy:gh-pages` in the docs folder. I didn't tested a full deploy run yet.

### CODE REVIEW

File: packages/docs/.gitignore [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+yarn-error.log*
+
+# local env files
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+# vercel
+.vercel
```
Comment (sibelius) [inline packages/docs/.gitignore:34]: do we need another gitignore here?

File: packages/docs/README.md [hunk 2]
```diff
@@ -1,21 +1,13 @@
-# DraftJS Plugins Docs
+This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
```
Comment (sibelius) [inline packages/docs/README.md:1]: maybe improve readm

**Review by sibelius — APPROVED** [review]
great stuff

File: packages/docs/.gitignore (hunk 1 above)
Comment (fxOne) [reply packages/docs/.gitignore:34]: You are right, I didn't saw this auto generated file

**Pull request merged**